src/pages/watchLater/WatchLater.jsx

```diff
@@ -0,0 +1,44 @@
+import { useState, useEffect } from "react";
+import { Link } from "react-router-dom";
+
+import "./watchLater.css";
+import { useDataContext } from "../../context/dataContext";
+
+export default function WatchLater() {
+  const { state } = useDataContext();
+
+  const [watchLater, setWatchLater] = useState([]);
+
+  const getWatchLaterData = () => state.watchLater;
```

Comment:
Watch Later can be fetched directly from `state`. No need of using a callback here..